include/deal.II/base/parsed_convergence_table_flags.h

```diff
@@ -0,0 +1,232 @@
+//-----------------------------------------------------------
+//
+//    Copyright (C) 2019 by the deal.II authors
+//
+//    This file is part of the deal.II library.
+//
+//    The deal2lkit library is free software; you can use it, redistribute
+//    it, and/or modify it under the terms of the GNU Lesser General
+//    Public License as published by the Free Software Foundation; either
+//    version 2.1 of the License, or (at your option) any later version.
+//    The full text of the license can be found in the file LICENSE at
+//    the top level of the deal2lkit distribution.
+//
+//-----------------------------------------------------------
+
+#ifndef dealii_base_parsed_convergence_table_flags_h
+#define dealii_base_parsed_convergence_table_flags_h
+
+#include <deal.II/base/config.h>
+
+#include <deal.II/base/patterns.h>
+
+DEAL_II_NAMESPACE_OPEN
+
+namespace ParsedConvergenceTableFlags
+{
+  /**
+   * Implemented Error norms.
+   */
+  enum NormType
+  {
+    none    = 0x00,
+    Linfty  = 0x01,
+    L2      = 0x02,
+    W1infty = 0x04,
+    H1      = 0x08,
+    custom  = 0x10,
+  };
```

Comment:
I agree. Let me change that to simply use VectorTools::NormType, and make sure we can parse VectorTools::NormType instead of creating a new type.

Reply:
Good idea. Maybe `VectorTools::parse_norm_type(const std::string &)`?

Reply:
I think it is enough to specialize `Patterns::Tools::Convert<VectorTools::NormType>`

Reply:
I've implemented `Patterns::Tools::Convert<VectorTools::NormType>`, and I'm now supporting all norm types supported by `VectorTools::integrate_difference`

Reply:
This seems hard to maintain because you define the enums in one place and the parsing in the other. Couldn't you move the parsing into a helper function in `namespace Vectortools`, and then just call this parsing function from `Convert<...>`?

Reply:
That's exactly what I did. This comment is below a file that does not exist anymore... I now have `Convert<VectorTools::NormType>` in `vector_tools.h`. It has to be in the namespace `Patterns::Tools` to be usable by `ParameterHandler::add_parameter`, but it is in the right file (`vector_tools.h`).

Reply:
Works for me.